jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/params/ParamResolverFactories.java

```diff
@@ -92,7 +94,15 @@ protected <T> ParamResolver<T> createParamResolver(Parameter parameter,
 
         BodyDeserializer<T> bodyDeserializer = deserializerFactory.getBodyDeserializer(paramType, consumesAnnotation);
         if (bodyDeserializer != null) {
-            return request -> Mono.just(deserializeBody(bodyDeserializer, request.getBody()));
+            return request -> {
+                T deserializedBody = deserializeBody(bodyDeserializer, request.getBody());
+
+                if (Objects.isNull(deserializedBody)) {
+                    return Mono.error(new WebException(HttpResponseStatus.BAD_REQUEST, "body.deserializer.error", "Deserialized body is null"));
+                }
+
+                return Mono.just(deserializeBody(bodyDeserializer, request.getBody()));
```

Comment:
You could just send in deserializedBody in instead of deserializing it once again.